eventsource-client/src/event_parser.rs

```diff
@@ -234,8 +230,13 @@ impl EventParser {
                             continue;
                         }
 
-                        self.last_event_id = value.to_string();
-                        event_data.id = self.last_event_id.clone();
+                        if value.is_empty() {
+                            self.last_event_id = Some("".to_string());
+                        } else {
```

Comment:
This is the key fix - if  `id` is present, but empty, then the server explicitly sent it and so it is `Some`.